src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1024,6 +1040,8 @@ query_path(const char *szInput, int *is_target_path, dfs_obj_t **parent, char *i
 
 			daos_inited = true;
 			atomic_fetch_add_relaxed(&daos_init_cnt, 1);
+			for (idx = 0; idx < low_fd_count; idx++)
+				close(low_fd_list[idx]);
```

Comment:
There are lots of error paths where these files would not be closed, you need to add extra error handling to this function.  Overall this function is large and would probably benefit from being split up.

Reply:
Thank you! Added a wrapper function as you suggested.